Modules/_io/bufferedio.c

```diff
@@ -2444,13 +2437,12 @@ static PyMethodDef bufferediobase_methods[] = {
 static PyType_Slot bufferediobase_slots[] = {
     {Py_tp_doc, (void *)bufferediobase_doc},
     {Py_tp_methods, bufferediobase_methods},
-    {Py_tp_traverse, bufferediobase_traverse},
     {0, NULL},
 };
 
 PyType_Spec bufferediobase_spec = {
     .name = "_io._BufferedIOBase",
-    .flags = (Py_TPFLAGS_DEFAULT | Py_TPFLAGS_BASETYPE | Py_TPFLAGS_HAVE_GC |
+    .flags = (Py_TPFLAGS_DEFAULT | Py_TPFLAGS_BASETYPE |
```

Comment:
Hmm, I am testing if this approach is better, so far it fails assertion on your script but the refleaks tests pass.

Reply:
I remembered all heap types should have GC. #87138

Reply:
> I remembered all heap types should have GC. https://github.com/python/cpython/issues/87138

The heap type still has GC but it now inheirts traverse and clear slots from its base because `Py_TPFLAGS_HAVE_GC` isn't set. See https://docs.python.org/3.12/c-api/gcsupport.html